### PR TITLE
Update client.lua

### PR DESCRIPTION
**Pull request by userMacieG**

### CODE REVIEW

**Review by Xinerki — COMMENTED** [review]
convince me to merge this

File: client.lua [hunk 1]
```diff
@@ -1,25 +1,39 @@
 CreateThread(function()
+	local playerPed = PlayerPedId()
```
Comment (Xinerki) [inline client.lua]: single non-updating player ped variable -> you will lose it the moment the model changes or player dies (which is inevitable)

File: client.lua [hunk 2]
```diff
@@ -1,25 +1,39 @@
 CreateThread(function()
+	local playerPed = PlayerPedId()
+	local boneId = 31086
+
 	RequestStreamedTextureDict("uhhh")
+
+	while not HasStreamedTextureDictLoaded("uhh")
```
Comment (Xinerki) [inline client.lua]: i dont care

File: client.lua [hunk 3]
```diff
@@ -1,25 +1,39 @@
...
+				local ratio = GetAspectRatio()
+				local scale = (1.0/dist) * (1.0/GetFinalRenderedCamFov()) * 12.0
+
 				SetDrawOrigin(pos)
 				DrawSprite("uhhh", IsPedAPlayer(v) and "canny" or "uncanny", 0.0, 0.0, scale, scale * ratio, 0.0, 255, 255, 255, 255)
 				ClearDrawOrigin()
 			end
 		end
+
+		Wait(0)
```
Comment (Xinerki) [inline client.lua:33]: why move it? i have my style i will not change

File: client.lua [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
 	end
-end)
\ No newline at end of file
+
+	SetStreamedTextureDictAsNoLongerNeeded("uhhh")
```
Comment (Xinerki) [inline client.lua:36]: this will never hit anyways

File: client.lua [hunk 5]
```diff
@@ -1,25 +1,39 @@
...
 			local pos = GetWorldPositionOfEntityBone(v, GetPedBoneIndex(v, boneId))
-			local dist = #(GetFinalRenderedCamCoord() - pos)
-			local scale = (1.0/dist) * (1.0/GetFinalRenderedCamFov()) * 12.0
+			local dist = #(camCoord - pos)
 			local visible = IsEntityVisible(v)
-			local los = HasEntityClearLosToEntity(PlayerPedId(), v, 17) or v == PlayerPedId()
+			local los = HasEntityClearLosToEntity(playerPed, v, 17) or v == playerPed
 			
 			if dist < 50.0 and los and visible then
+				local ratio = GetAspectRatio()
```
Comment (Xinerki) [inline client.lua:24]: a) why put this inside the if statement?
b) why do you not get this once, like the playerped? this one is much more likely to remain the same throughout a session.

File: client.lua [hunk 6]
```diff
@@ -1,25 +1,39 @@
...
+	while true do
 		local pool = GetGamePool("CPed")
-		table.sort(pool, function(a, b) return #(GetEntityCoords(a) - GetFinalRenderedCamCoord()) > #(GetEntityCoords(b) - GetFinalRenderedCamCoord()) end)
+		local camCoord = GetFinalRenderedCamCoord()
+
+		table.sort(pool, function(a, b) return #(GetEntityCoords(a) - camCoord) > #(GetEntityCoords(b) - camCoord) end)
 		
 		for i,v in pairs(pool) do
-			local boneId = 31086
+			playerPed = PlayerPedId()
```
Comment (Xinerki) [inline client.lua]: you update it afterall, so - why bother?

Comment (userMacieG) [reply client.lua]: imo its still better to make that variable outside of while loop, also i fucked up and put that insidie for loop, fixed

File: client.lua (hunk 5 above)
Comment (userMacieG) [reply client.lua:24]: a) we dont use ratio variable before dist, los and visible check so there is no reason to create local and call that native
b) but still someone can change thier aspect ratio whille playing

File: client.lua (hunk 4 above)
Comment (userMacieG) [reply client.lua:36]: true, but why not

File: client.lua (hunk 3 above)
Comment (userMacieG) [reply client.lua:33]: after starting script we want to draw that in current frame not next, also better practice overall

File: client.lua (hunk 2 above)
Comment (userMacieG) [reply client.lua]: good for you

Comment (userMacieG) [reply client.lua]: also forgot `do` so fixed

File: client.lua (hunk 1 above)
Comment (userMacieG) [reply client.lua]: thats why i updated that anyway, moved to loop

**Comment by userMacieG** [comment]
overall i was bored when talking with someone and saw that script, so idk you can close

**Pull request closed without merging**